src/prometheus_server.py

```diff
@@ -34,13 +37,25 @@ def reload_configuration(self) -> bool:
           True if reload succeeded (returned 200 OK); False otherwise.
         """
         url = f"{self.base_url}/-/reload"
+        errors = list(range(400, 452)) + list(range(500, 512))
```

Comment:
These HTTP status codes feel like magic numbers. Maybe a comment here would be worthwhile. Perhaps a more descriptive variable names such as `http_errors` may would also be more revealing.